flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -2013,23 +2077,25 @@ private void abortPendingCheckpoints(
 
     private void rescheduleTrigger(long tillNextMillis) {
         cancelPeriodicTrigger();
+        nextCheckpointTriggeringRelativeTime = clock.relativeTimeMillis() + tillNextMillis;
         currentPeriodicTrigger = scheduleTriggerWithDelay(tillNextMillis);
     }
 
     private void cancelPeriodicTrigger() {
         if (currentPeriodicTrigger != null) {
+            nextCheckpointTriggeringRelativeTime = Long.MAX_VALUE;
             currentPeriodicTrigger.cancel(false);
             currentPeriodicTrigger = null;
         }
     }
 
     private long getRandomInitDelay() {
-        return ThreadLocalRandom.current().nextLong(minPauseBetweenCheckpoints, baseInterval + 1L);
+        return ThreadLocalRandom.current()
+                .nextLong(minPauseBetweenCheckpoints, getCurrentCheckpointInterval() + 1L);
```

Comment:
Given that `backlogOperators` must be empty at the time `getRandomInitDelay()` is invoked, would it be simpler to still use `baseInterval + 1L`?